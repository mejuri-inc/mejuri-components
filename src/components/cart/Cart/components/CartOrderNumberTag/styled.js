import styled from 'styled-components'
import colors from 'styles/colors'

export const Wrapper = styled.div`
  text-align: center;
  margin-top: 5px;
`

export const OrderNumber = styled.p.attrs({
  'data-h': 'order-number',
  'data-testid': 'order-number'
})`
  display: inline-block;
  margin: 0;
  padding: 5px;
  font-weight: bold;
  background: ${colors.beige};
  color: ${colors.white};
`
