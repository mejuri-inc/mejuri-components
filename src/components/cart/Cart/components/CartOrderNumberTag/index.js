import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { CartOrderNumberWrapper, OrderNumberParagraph } from './styled'

export class CartOrderNumberTag extends PureComponent {
  render() {
    return (
      <CartOrderNumberWrapper>
        <a
          href={'/shop/admin/orders/' + this.props.orderNumber + '/edit'}
          rel='noreferrer'
          target='_blank'
        >
          <OrderNumberParagraph>{this.props.orderNumber}</OrderNumberParagraph>
        </a>
      </CartOrderNumberWrapper>
    )
  }
}

CartOrderNumberTag.propTypes = {
  orderNumber: PropTypes.string
}

export default CartOrderNumberTag
