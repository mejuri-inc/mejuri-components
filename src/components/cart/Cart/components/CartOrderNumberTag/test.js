import React from 'react'
import { render } from '@testing-library/react'
import CartOrderNumberTag from './index'

describe('CartOrderNumberTag', () => {
  it('Renders with the order number provided', () => {
    const { container } = render(
      <CartOrderNumberTag orderNumber={'M62723604QU'} />
    )

    expect(container).toMatchSnapshot()
  })
})