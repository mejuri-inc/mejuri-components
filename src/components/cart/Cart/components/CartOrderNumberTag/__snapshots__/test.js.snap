// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartOrderNumberTag Renders with the order number provided 1`] = `
<div>
  <div
    class="sc-bdnylx iIEgSZ"
  >
    <a
      href="/shop/admin/orders/M62723604QU/edit"
      rel="noreferrer"
      target="_blank"
    >
      <div
        class="sc-gtssRu fnkEfa"
      >
        M62723604QU
      </div>
    </a>
  </div>
</div>
`;
